pkg/cmd/cli/sa/gettoken.go

```
@@ -0,0 +1,129 @@
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/serviceaccounts"
+)
+
+const (
+	GetServiceAccountTokenRecommendedName = "get-token"
+
+	getServiceAccountTokenShort = `Get a token assigned to a service account.`
+
+	getServiceAccountTokenLong = `
+Get a token assigned to a service account.
+
+If the service account has multiple tokens, the first token found will be returned.
+
+Service account API tokens are used by service accounts for privileged access to the API. If actions that cannot
+be accomplished by OpenShift client actions using the service account are required, and the actions necessiate 
+privileged requests to the API server, the service account token is necessary for authorization. This command will
+get the service account token from a service account to facilitate these external actions.
+`
+
+	getServiceAccountTokenUsage = `%s SA-NAME`
+
+	getServiceAccountTokenExamples = `  # Get the service account token from service account 'custom-sa'
+  $ %[1]s 'custom-sa'
+`
+)
+
+type GetServiceAccountTokenOptions struct {
+	SAName        string
+	Namespace     string
+	SAClient      unversioned.ServiceAccountsInterface
+	SecretsClient unversioned.SecretsInterface
+
+	Out io.Writer
+	Err io.Writer
+}
+
+func NewCommandGetServiceAccountToken(name, fullname string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &GetServiceAccountTokenOptions{
+		Out: out,
+		Err: os.Stderr,
+	}
+
+	getServiceAccountTokenCommand := &cobra.Command{
+		Use:     fmt.Sprintf(getServiceAccountTokenUsage, name),
+		Short:   getServiceAccountTokenShort,
+		Long:    getServiceAccountTokenLong,
+		Example: fmt.Sprintf(getServiceAccountTokenExamples, fullname),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := options.Complete(args, f, cmd); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(cmd, err.Error()))
```

That actually masks the error translation we want for some kinds of errors.  Make complete return a usage error in the cases where it makes sense and simply run `CheckErr` on the result.